### ISSUE TITLE
GetBucketLocation fails for any bucket in us-east-1 (v3)

### ISSUE THREAD

**Issue by rausnitz** [soto-project/soto#502]
**Describe the bug**
When I send a request to GetBucketLocation for a bucket in region `us-east-1`, I get the following error:
```
▿ ⚠️ DecodingError: Cannot initialize BucketLocationConstraint from invalid String value 
- id: DecodingError.dataCorrupted

  ▿ dataCorrupted : Context
    ▿ codingPath : 1 element
      - 0 : CodingKeys(stringValue: "LocationConstraint", intValue: nil)
    - debugDescription : "Cannot initialize BucketLocationConstraint from invalid String value "
    - underlyingError : nil
```

According to the AWS [docs](https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetBucketLocation.html#API_GetBucketLocation_ResponseSyntax), `Buckets in Region us-east-1 have a LocationConstraint of null.` 

**To Reproduce**
```
public static func getBucketLocation() -> EventLoopFuture<S3.GetBucketLocationOutput> {
    let request = S3.GetBucketLocationRequest(bucket: "some-us-east-1-bucket")
    return S3().getBucketLocation(request)
}
```

**Expected behavior**
The function should return a `GetBucketLocationOutput ` where `locationConstraint` is `nil`, instead of failing to decode the `GetBucketLocationOutput`.

**Setup (please complete the following information):**
 - OS: macOS
 - Version of soto: 3.5.2
 - Authentication mechanism: credentials

**Additional context**
n/a

**Comment by adam-fowler** [comment]
I'm assuming you can't upgrade to v4.0. I'm not really supporting v3.x anymore.

Adding `us-east-1` to this enum should fix the issue https://github.com/soto-project/soto/blob/08d3c295049cca3494203125277285ef223189cd/Sources/AWSSDKSwift/Services/S3/S3_Shapes.swift#L336-L339

If you want to make a PR to `3.x.x` branch please do so. I can make a release with the change in after

**Comment by rausnitz** [comment]
Yeah I figured v3 isn't getting updates.

In any case I don't think your suggestion of adding `us-east-1` to that enum would fix the issue. See above: `Buckets in Region us-east-1 have a LocationConstraint of null.`

I tested this in a few regions.

When the bucket is in us-west-2, the value of `node` on AWSClient.swift line 565 is:
```
<BucketLocation><LocationConstraint xmlns="http://s3.amazonaws.com/doc/2006-03-01/">us-west-2</LocationConstraint></BucketLocation>
```

When the bucket is in us-east-1, the value of `node` on AWSClient.swift line 565 is:
```
<BucketLocation><LocationConstraint xmlns="http://s3.amazonaws.com/doc/2006-03-01/"></LocationConstraint></BucketLocation>
```

These XML results are both expected. However in the latter case, the decoded `GetBucketLocationOutput.locationConstraint` value _should_ be `nil` but instead there is a decoding error.
